index.js

```diff
@@ -1,26 +1,165 @@
 const core = require('@actions/core');
 const github = require('@actions/github');
+const list = require('markdown-list');
+
+export function formatChType(type) {
+  const featureRegex = /(feat+)/;
+  if (!type) {
+    return 'other';
+  }
+
+  if (type.match(featureRegex)) {
+    return 'feature';
+  }
+
+  return type.replace(/[^a-zA-Z ]/g, '');
+}
+
+export function formatCommits(commits, chStoryUrl) {
+  const {
+    payload: {
+      repository: { url: repoUrl },
+    },
+  } = github.context;
+
+  return commits.reduce((acc, com) => {
+    const {
+      commit: { message },
+      sha: origSha,
+    } = com;
+    const regex = /(?<chType>\(\w*\))?(\s)?(?<prMsg>\w*\W*.+?)(\s+)?(\[)?(ch)?(?<chId>\d+)?(\])?\s\(#(?<prNumber>\d+)\)/;
+    const matches = message.match(regex);
+    const chType = formatChType(matches && matches.groups.chType);
+    const prMsg = (matches && matches.groups.prMsg) || message;
+    const prLink =
+      (matches &&
+        matches.groups.prNumber &&
+        `[#${matches.groups.prNumber}](${repoUrl}/pull/${matches.groups.prNumber})`) ||
+      null;
+    const chLink =
+      (matches &&
```

Comment:
If it doesn't check for matches first, then it's going to error when that group doesn't exist. So it's just a safe guard.

Reply:
Sort of reflecting on @kuritz confusing when reviewing.  These statements would be easier to follow as ternary rather than logical chaining.

```
const chLink = matches && matches.groups.chId
	? `[ch${matches.groups.chId}](${chStoryUrl}/${matches.groups.chId})`
	: null;
```